builtin/providers/cidr/datasource_cidr_subnet.go

```diff
@@ -0,0 +1,108 @@
+package cidr
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceSubnet() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSubnetRead,
+
+		Schema: map[string]*schema.Schema{
+			"cidr_block": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The CIDR Block for the entire network (aka supernet)",
+				Required:    true,
+			},
+			"start_after": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The subnet in CIDR notation to offset subnet creation by",
+				Optional:    true,
+			},
+			"subnet_count": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The number of subnets to create",
+				Default:     1,
+				Optional:    true,
+			},
+			"subnet_mask": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The deisred subnet mask to use for creation",
+				Required:    true,
+			},
+			"max_subnet": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"subnet_cidrs": &schema.Schema{
+				Type:        schema.TypeSet,
+				Description: "The set of subnets in CIDR notation",
+				Optional:    true,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataSourceSubnetRead(d *schema.ResourceData, meta interface{}) error {
+	d.SetId(subnetId(d))
+	subnetCount := d.Get("subnet_count").(int)
+	subnets := make([]*net.IPNet, subnetCount)
+	subnetCIDRs := make([]string, subnetCount)
+	mask := d.Get("subnet_mask").(int)
+
+	_, startNet, perr := net.ParseCIDR(d.Get("cidr_block").(string))
+	if perr != nil {
+		return fmt.Errorf("cidr_block parse error %v\n", perr)
+	}
+
+	var currentSubnet *net.IPNet
+	startAfter, sap := d.GetOk("start_after")
+	if sap {
+		_, offsetSubnet, perr := net.ParseCIDR(startAfter.(string))
+		if perr != nil {
+			return fmt.Errorf("start_after %v resulted in pasre error %v\n", startAfter, perr)
```

Comment:
parse

Reply:
fixed in https://github.com/moofish32/terraform-provider-cidr/pull/2